src/wallet/wallet.cpp

```diff
@@ -2329,6 +2329,13 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                     if (dPriority >= dPriorityNeeded && AllowFree(dPriority))
                         break;
                 }
+                else if (fSendFreeTransactions && LimitFree())
+                {
+                    strFailReason =
+                        _("You can not send free transactions if you have configured a -limitfreerelay of zero");
+                    return false;
+                }
```

Comment:
What if the zero fee txns is created using "old" utxo so that the resulting txns could be considered an high priority one?

Reply:
This is just handling a rare edge case.  If you have configured limitfreerelay of zero..that means you're not going to nor intend to route "any" free transactions...so if you created one with your wallet , it won't get routed by your very own node and this is just warning that you'll need to adjust your limitfreerelay if you want to send free transactions.  Priority doesn't matter in this context.   But now that you mention it, I should break that out into it's own "if" statement rather than an else if...i'll change that.

Reply:
this is great.  I think from a usability perspective, we should not allow a tx to be created that our own node would normally reject.  This will stop people from accidentally making zombie tx.